### PR TITLE
Initial support of pwn.college-side mac support

### PR DESCRIPTION
**Pull request by adamdoupe**

### CODE REVIEW

File: nginx-proxy/etc/nginx/vhost.d/default_location [hunk 1]
```diff
@@ -8,3 +8,5 @@ location /internal/ {
 
     proxy_ignore_headers X-Accel-Redirect;
 }
+proxy_read_timeout 600s;
```
[the remark this answers is not in the document]
Comment (ConnorNelson) [reply nginx-proxy/etc/nginx/vhost.d/default_location:11]: I'm also a little bit nervous. Are you expecting that starting a mac challenge is going to hang the thread for a minute+? This feels like a bad design which may make me even more in favor of starting a linux container, and they connect to the mac with `mac start`, where this entire feature lives in the challenge container.

The user connects in and does `macos connect`, just like they currently do `vm connect` for linux and `windows connect` for windows. Your macos server will need to be able to auth the requests, which is something we can easily achieve. I think we'll just need the dojo to public key sign the user_id/challenge_id, your macos server verifies this, and we open up a static IP in the workspace firewall. 

Of course we can worry about moving to this simpler design after your challenges launch.

**Review by ConnorNelson — COMMENTED** [review]
TLDR: probably no _critical_ changes to launch (I think redis.get/set is thread-safe for basic operations from my reading), but I would short-medium term like to move all of this logic into the challenge image. We should be able to *completely* revert this PR. It's honestly challenge logic, which is much easier to keep maintenance responsibilities out of the dojo codebase. But for now YOLO.

File: docker-compose.yml [hunk 2]
```diff
@@ -105,6 +105,10 @@ services:
       - VIRTUAL_HOST=${VIRTUAL_HOST}
       - VIRTUAL_PORT=8000
       - LETSENCRYPT_HOST=${LETSENCRYPT_HOST}
+      - MAC_HOSTNAME=${MAC_HOSTNAME}
+      - MAC_USERNAME=${MAC_USERNAME}
+      - MAC_KEY_FILE=${MAC_KEY_FILE}
+      - MAC_GUEST_CONTROL_FILE=${MAC_GUEST_CONTROL_FILE}
```
[the remark this answers is not in the document]
Comment (adamdoupe) [reply docker-compose.yml:111]: This makes testing on different systems easier (my laptop vs. the actual server where this will run).
